### ISSUE TITLE
[2022 CVPR Oral] Scaling Vision Transformers to Gigapixel Images via Hierarchical Self-Supervised Learning

### ISSUE THREAD

**Issue by Jasonlee1995** [Jasonlee1995/AI_Papers#82]
병리 dataset에서는 대부분 WSI 레벨의 annotation이 있는데, 이는 gigapixel image이기에 기존의 비전에서 방법을 사용하기 어려움

기존의 경우 MIL (multiple instance learning) 방식을 사용하였는데, 이에서 벗어나 hierarchical하게 representation을 배워 WSI의 representation을 구하는게 논문의 핵심

논문에서 중요한 부분만 요약

### 1.1. Problem Formulation

WSI 레벨 annotation이 있는데 이는 natural image dataset과 다음과 같은 2가지 점으로 인해 직접적인 적용이 어려움

1. Gigapixel image
2. WSI마다 image width, height가 제각각 다름

저자들은 vision transformer를 사용하여 2번 문제를 해결할 수 있으며, hierarchical한 방식을 사용하여 1번의 문제를 해결

### 1.2. Hierarchical Image Pyramid Transformer (HIPT) Architecture
<img width="1061" alt="image" src="https://user-images.githubusercontent.com/49643709/174530557-d72c4701-1627-4f4c-83c3-e387bf3d24b3.png">
<img width="650" alt="image" src="https://user-images.githubusercontent.com/49643709/174530678-834e78db-2043-4a66-9026-bc4355653f98.png">

전체적인 flow는 다음과 같음

1. 256 x 256 image representation 구하기 : 16 x 16 image patch, 16 x 16 patch 개수 (16 x 16 image patch linear embedding)
2. 4096 x 4096 image representation 구하기 : 256 x 256 image patch, 16 x 16 patch 개수 (1번에서의 network를 이용하여 256 x 256 image patch embedding)
3. WSI image representation 구하기 : 4096 x 4096 image patch, ? x ? patch 개수 (2번에서의 network를 이용하여 4096 x 4096 image patch embedding)

`CLS`라는 것이 보이는데, 예시를 들자면 256 x 256 image의 representation을 256 x 256 image를 학습시킨 network의 `CLS` token의 representation으로 보겠다는 것임
(not global average pooling)

#### 1.2.1. `ViT 256-16` for Cell-Level Aggregation

`8 layers, 6 heads, 384 hidden dimension`

Embed 256 x 256 image using 16 x 16 image patch

16 x 16 image patch를 384 dimension으로 linear embedding

#### 1.2.2. `ViT 4096-256` for Patch-Level Aggregation

`4 layers, 6 heads, 192 hidden dimension`

Embed 4096 x 4096 image using 256 x 256 image patch

256 x 256 image patch는 pre-trained `ViT 256-16`를 이용하여 embedding

`ViT 256-16`의 embedding dimension은 384인데 어떻게 192 dimension으로 만드는지?에 대해서 코드를 보면 256 x 256 image patch를 384 dimension으로 만들고 `self.phi`를 이용해서 192 dimension으로 줄임

<img width="1095" alt="image" src="https://user-images.githubusercontent.com/49643709/174555147-5fa0c010-4064-4a0d-a7a1-2b18b94863ba.png">

#### 1.2.3. `ViT WSI-4096` for Patch-Level Aggregation

`2 layers, 3 heads, 192 hidden dimension`

Embed WSI image using 4096 x 4096 image patch

주목할만한 점은 4096 x 4096에서의 potential tissue segmentation irregularity로 인해 positional embedding을 무시한다고 함
(due to potential tissue segmentation irregularities in patching at [4096×4096], we ignore positional embeddings at this stage)

### 1.3. Hierarchical Pretraining

1. `ViT 256-16` pretrain using DINO
    - 256 x 256 image
    - 96 x 96 8 local views + 224 x 224 2 global views
2. Freeze `ViT 256-16` and pretrain `ViT 4096-256` using DINO
    - 16 x 16 vector
    - 6 x 6 8 local views + 14 x 14 2 global views

## 2. Experiments

- Dataset : 10,678 FFPE H&E-stained diagnostic slides from 33 cancer types in the TCGA
    - 20x objective
    - 408,218 4096 x 4096 images
    - 104M 256 x 256 images
- Pretrain
    - `ViT 256-16` : 400,000 iterations, AdamW optimizer, 256 batch size, 0.0005 lr with 10 epochs warm-up, cosine decay
    - `ViT 4096-256` : 200,000 iterations, similar as `ViT 256-16` train setting
- Finetune
    - Freeze `ViT 256-16` and `ViT 4096-256` and finetune only `ViT WSI-4096`
    - 20 epochs, Adam optimizer, 1 batch size with 32 gradient accumulation, 0.01 learning rate
    - Use survival cross-entropy loss on survival prediction

### 2.1. Slide-Level Classification
<img width="501" alt="image" src="https://user-images.githubusercontent.com/49643709/174564476-cb8005dd-793f-4a65-ae74-6de2ca48a65c.png">
<img width="1064" alt="image" src="https://user-images.githubusercontent.com/49643709/174564563-09e6ff1a-14b1-4776-8a65-d4d8a232d322.png">

### 2.2. Survival Prediction
<img width="503" alt="image" src="https://user-images.githubusercontent.com/49643709/174564715-8e100990-ccbc-4532-9125-8e4457041383.png">
<img width="1043" alt="image" src="https://user-images.githubusercontent.com/49643709/174564776-d6fbffc9-8b9c-4a5e-8c33-e52ddc6e8cc8.png">

### 2.3. Self-Supervised ViTs Find Unique Morphological Phenotypes

각 모델들의 last ViT stage의 각 head의 attention map을 lineary interpolate하여 visualize

<img width="546" alt="image" src="https://user-images.githubusercontent.com/49643709/174565074-f50057e0-7b23-4896-b1a5-fe09c8984e5e.png">

`ViT 256-16` captures three distinct fine-grained morphological phenotypes

1. General stroma tissue and red blood cells (h = 1, 2)
2. Cells : normal, atypical, lymphocyte (h = 3, 4)
3. White spaces : luminal spaces, fat regions, air pockets (h = 5, 6)

<img width="1105" alt="image" src="https://user-images.githubusercontent.com/49643709/174565782-66dd4e0f-58da-4271-92ae-76badadb55b3.png">

`ViT 4096-256` captures two distinct coarse-grained phenotypes

1. Tumorstroma interface (h = 1, 2, 3)
2. Nested tumor cells and other high tumor cellularity regions (h = 4, 5, 6)

`ViT 256-16` attention maps capture only nuclear features (e.g. - nuclear atypia, shape and size of cells)

`ViT 4096-256` attention maps are able to model the patterns of nested tumor growth, tumor invasion into fat and stroma regions, and other tissue-to-tissue relationships

## 3. Supplement
<img width="544" alt="image" src="https://user-images.githubusercontent.com/49643709/174566956-33b14e11-3eaf-42e7-b059-74bb9d9c0498.png">
<img width="546" alt="image" src="https://user-images.githubusercontent.com/49643709/174567012-8b9e7514-7a34-4d0c-850b-d07c944ad4ef.png">
<img width="1123" alt="image" src="https://user-images.githubusercontent.com/49643709/174567159-b98659a3-9789-4ee8-bdc9-035b659d7150.png">
<img width="1100" alt="image" src="https://user-images.githubusercontent.com/49643709/174567199-e3008f48-5962-4aa8-a841-2002867be58d.png">
<img width="1100" alt="image" src="https://user-images.githubusercontent.com/49643709/174567242-9266b23f-bef3-459f-86b8-7dc0fcdc5f6f.png">
